### PR TITLE
Law status

### PR DESCRIPTION
**Pull request by svamaa**

### CODE REVIEW

File: pyramid_oereb/lib/sources/plr.py [hunk 1]
```diff
@@ -54,8 +54,8 @@ def __init__(self, **kwargs):
             source (dict): The configuration dictionary of the public law restriction
             hooks (dict of str): The hook methods: get_symbol, get_symbol_ref. They have to be provided as
                 dotted string for further use with dotted name resolver of pyramid package.
-            law_status (dict of str): The multiple match configuration to provide more flexible use of the
-                federal specified classifiers 'inForce' and 'runningModifications'.
+            law_status (dict of str): The text of the law status. It must be from code "inKraft" or
```
Comment (voisardf) [inline pyramid_oereb/lib/sources/plr.py:57]: I'm not quite sure to understand this comment? 🙈 What do you mean to say with the second part?

File: pyramid_oereb/standard/models/main.py [hunk 2]
```diff
@@ -192,6 +192,23 @@ class ExclusionOfLiability(Base):
     content = sa.Column(JSONType, nullable=False)
 
 
+class LawStatus(Base):
+    """
+    The container where you can throw in all the law status texts this application
+    should have access to, for creating extracts.
+    Attributes:
+        id (int): The identifier. This is used in the database only and must not be set manually. If
```
Comment (voisardf) [inline pyramid_oereb/standard/models/main.py:200]: do we need the id? Should the code not be a primary key already?

File: pyramid_oereb/standard/pyramid_oereb.yml.mako [hunk 3]
```diff
@@ -859,8 +872,19 @@ pyramid_oereb:
...
+      # Example of how sub_themes sorting can be activated (uncomment to enable):
+      #sub_themes:
+      #  sorter:
+      #    module: pyramid_oereb.contrib.print_proxy.sub_themes.sorting
+      #    class_name: ListSort
+      #    params:
+      #      list:
+      #        - SubTheme3
+      #        - SubTheme2
+      #        - SubTheme1
```
Comment (voisardf) [inline pyramid_oereb/standard/pyramid_oereb.yml.mako:887]: This part should be moved to the Land Use Plan section as it is basically the only one for now that allows subthemes.
But I thought Cécile already moved it in an earlier PR? (please check and delete or move this comment)

File: pyramid_oereb/standard/sources/plr.py [hunk 4]
```diff
@@ -38,8 +37,8 @@ def __init__(self, **kwargs):
             source (dict): The configuration dictionary of the public law restriction
             hooks (dict of str): The hook methods: get_symbol, get_symbol_ref. They have to be provided as
                 dotted string for further use with dotted name resolver of pyramid package.
-            law_status (dict of str): The multiple match configuration to provide more flexible use of the
-                federal specified classifiers 'inForce' and 'runningModifications'.
+            law_status (dict of str): The text of the law status. It must be from code "inKraft" or
```
Comment (voisardf) [inline pyramid_oereb/standard/sources/plr.py:40]: same as above: I do not get exactly the meaning of this description.
Do you want to say it has to be one out of the three "inKraft", "AenderungMitVorwirkung" or "AenderungOhneVorwirkung" or is the emphasis on something else?

File: pyramid_oereb/standard/sources/plr.py [hunk 5]
```diff
@@ -343,7 +331,8 @@ def get_document_records(self, params, public_law_restriction_from_db):
             article_nrs = legal_provision.article_numbers.split('|') if legal_provision.article_numbers \
                 else None
             article_numbers.append(article_nrs)
-        return self.from_db_to_document_records(documents_from_db, article_numbers)
+        document_records = self.from_db_to_document_records(documents_from_db, article_numbers)
```
Comment (voisardf) [inline pyramid_oereb/standard/sources/plr.py:334]: nice cleanup catch! 👍

File: sample_data/law_status.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
+[
+    {
+        "id": 1,
```
Comment (voisardf) [inline sample_data/law_status.json:3]: same question as above - should we keep the id as the code is already an identifier?

**Review by voisardf — COMMENTED** [review]
Nice work, thanks! 
I have some questions for clarification and a few typos/spelling errors I found. I have not had the time yet to test the branch locally.

**Comment by vvmruder** [comment]
I will review when tests are finished

File: pyramid_oereb/standard/models/main.py (hunk 2 above)
Comment (svamaa) [reply pyramid_oereb/standard/models/main.py]: I removed the Id already. Good issue ;)

File: pyramid_oereb/standard/pyramid_oereb.yml.mako (hunk 3 above)
Comment (svamaa) [reply pyramid_oereb/standard/pyramid_oereb.yml.mako:887]: I probably accidentally copied this part here. Just removed it

**Comment by voisardf** [comment]
@svamaa : I have still seven failing tests for this PR - is that as it should be? Or is there still work in progress?

```
================================================================================================== short test summary info =====
FAILED tests/renderer/test_xml.py::test_extract_against_schema[parameter0-test_extract0] - AttributeError: 'str' object has no attribute 'content'
FAILED tests/renderer/test_xml.py::test_extract_against_schema[parameter1-test_extract1] - AttributeError: 'str' object has no attribute 'content'
FAILED tests/renderer/test_xml.py::test_extract_against_schema[parameter2-test_extract2] - AttributeError: 'str' object has no attribute 'content'
FAILED tests/renderer/xml/test_public_law_restriction.py::test_sub_theme - AttributeError: 'str' object has no attribute 'type'
FAILED tests/xtf_import/test_legend_entry.py::test_parse - TypeError: 'topic' is an invalid keyword argument for LegendEntry
FAILED tests/xtf_import/test_public_law_restriction.py::test_parse - TypeError: 'legend_text' is an invalid keyword argument for PublicLawRestriction
FAILED tests/xtf_import/test_federal_topic.py::test_load - SystemExit: 1
================================================================================ 7 failed, 363 passed, 43 warnings in 159.61s (0:02:39) =====
```

**Comment by vvmruder** [comment]
> @svamaa : I have still seven failing tests for this PR - is that as it should be? Or is there still work in progress?
> 
> ```
> ================================================================================================== short test summary info =====
> FAILED tests/renderer/test_xml.py::test_extract_against_schema[parameter0-test_extract0] - AttributeError: 'str' object has no attribute 'content'
> FAILED tests/renderer/test_xml.py::test_extract_against_schema[parameter1-test_extract1] - AttributeError: 'str' object has no attribute 'content'
> FAILED tests/renderer/test_xml.py::test_extract_against_schema[parameter2-test_extract2] - AttributeError: 'str' object has no attribute 'content'
> FAILED tests/renderer/xml/test_public_law_restriction.py::test_sub_theme - AttributeError: 'str' object has no attribute 'type'
> FAILED tests/xtf_import/test_legend_entry.py::test_parse - TypeError: 'topic' is an invalid keyword argument for LegendEntry
> FAILED tests/xtf_import/test_public_law_restriction.py::test_parse - TypeError: 'legend_text' is an invalid keyword argument for PublicLawRestriction
> FAILED tests/xtf_import/test_federal_topic.py::test_load - SystemExit: 1
> ================================================================================ 7 failed, 363 passed, 43 warnings in 159.61s (0:02:39) =====
> ```

When I run it, (`make clean-all test`) then 83 tests are failing

**Review by vvmruder — APPROVED** [review]
Looks good now

**Comment by marionb** [comment]
NOTE:
this test `tests/contrib/print_proxy/test_mapfish_print.py::test_mapfish_print_entire_extract` is failing because now some elements in the json are multilingual dictionaries. The print json is expecting only a string value instead. To fix this the print proxy needs to be adapted to receive a multilingual dictionary and pass the expected language in the json spec passed to the templates.
This will be treated in another issue.

**Pull request merged**